### PR TITLE
WIP: modularizing docs content

### PR DESCRIPTION
**Pull request by jenmalloy**
Do not merge yet

### CODE REVIEW

File: documentation/common/install-procedure.adoc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 .Procedure 
 
-. Download one of the releases from https://github.com/EnMasseProject/enmasse/releases and unpack it.
+* Download one of the releases from the link:https://github.com/EnMasseProject/enmasse/releases[GitHub repository] and unpack it.
```
Comment (ncbaratta) [inline documentation/common/install-procedure.adoc]: You're going to want to turn this link into an attribute.

File: documentation/service_admin/installing-manual-singletenant.adoc [hunk 2]
```diff
@@ -51,10 +53,8 @@ oc create -f ./resources/address-space-controller/deployment.yaml
 ----
 +
 The deployments required for running {ProductName} are now created.
-This guide will walk through the manual deployment procedure and should work on any platform
-supporting the OpenShift client.
 
-. {ProductName} will be up and running once all pods in the 'enmasse' namespace are in the `Running` state:
+. {ProductName} is running once all pods in the `enmasse` namespace are in the `Running` state:
```
Comment (ncbaratta) [inline documentation/service_admin/installing-manual-singletenant.adoc]: Will this always be enmasse? If not we should add an attribute.

File: documentation/service_admin/installing-openshift.adoc [hunk 3]
```diff
@@ -9,41 +9,26 @@ include::common/prerequisites-openshift.adoc[leveloffset=+1]
...
-|keycloak_admin_password|The admin password to use for the standard authentication service keycloak instance. |- |yes (if standard is enabled)
-|keycloak_http_url|The OAUTH URL to use for keycloak.| standard authentication service ip |no
-|api_server|Enable REST API server.|true |no
-|register_api_server|Register API server with Kubernetes master.|false |no
-|secure_api_server|Enable mutual TLS for API server.|false |no
-|===
-
-An example inventory that enables both the API server and Service Broker integration:
+Installing {ProductName} using Ansible requires creating an inventory file with the variables for configuring the
+system. Example inventory files can be found in the `ansible/inventory` folder. For more information about the supported Ansible configuration settings see xref:ref_ansible-config-settings_{context}[Ansible configuration settings].
```
Comment (ncbaratta) [inline documentation/service_admin/installing-openshift.adoc]: I don't think you want 'context' in this link - instead, you want whatever the context is on the page you're linking to.

File: documentation/common/proc-install-procedure.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+// assembly-installing-openshift.adoc
+
+include::attributes.adoc[]
+
+[id='downloading_{context}']
+= Downloading {ProductName}
+
+.Procedure 
+ifdef::Downloading[]
+* Download one of the releases from the link:https://github.com/EnMasseProject/enmasse/releases[GitHub repository] and unpack it.
```
Comment (ncbaratta) [inline documentation/common/proc-install-procedure.adoc]: ^ on link

Comment (ncbaratta) [reply documentation/common/proc-install-procedure.adoc]: Should we say 'the most recent release' instead of 'one of the releases'?

File: documentation/common/snip-prerequisites-openshift.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
+:context: enmasse
```
Comment (ncbaratta) [inline documentation/common/snip-prerequisites-openshift.adoc]: I think ... and we need to talk about this as a team ... you want the content to be in the file that includes the snip.  You want to define the context before you include it so that you can use it multiple times with different context each time. If the context is defined in the file, then it's always the same and can't be reused without throwing build errors because of duplicate IDs.

File: documentation/common/snip-prerequisites-openshift.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+.Prerequisites
+
+* To install {ProductName}, the OpenShift client tools are required. You can download the OpenShift
+Origin client from link:https://github.com/openshift/origin/releases[OpenShift Origin^]. {ProductName} has
```
Comment (ncbaratta) [inline documentation/common/snip-prerequisites-openshift.adoc]: This is only for the upstream.

File: documentation/service_admin/assembly-installing-openshift.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+= Installing {ProductName} on OpenShift
+
+{ProductName} can be installed using automated link:https://www.ansible.com/[Ansible] playbooks, the
+`deploy.sh` script, or the manual steps.
+
+*NOTE*: You can invoke the deployment script with `-h` to view a list of options.
+
+include::common/snip-prerequisites-openshift.adoc[]
+
+:parent-context: {context}
```
Comment (ncbaratta) [inline documentation/service_admin/assembly-installing-openshift.adoc]: Why is this here? I'd put it at the top right above the attributes.

File: documentation/service_admin/proc-creating-project.adoc [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create namespace enmasse
+----
+
+. Set the `enmasse` namespace as the default namespace:
++
+[options="nowrap",subs="attributes"]
+----
+kubectl config set-context $(kubectl config current-context) --namespace=enmasse
```
Comment (ncbaratta) [inline documentation/service_admin/proc-creating-project.adoc]: Why isn't this one using the cmdcli attribute?  Also, there might be a better way to handle this, let me think on it.

File: documentation/service_admin/proc-deploying-address-space-controller.adoc [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+include::common/attributes.adoc[]
+
+:cmdcli: oc
+
+[id='deploying-address-space-controller_{context}']
+= Deploying the address space controller
+
+The address space controller is responsible for creating the infrastructure used by address spaces.
+
+*NOTE*: To install {ProductName} on OpenShift, you must have cluster-admin access to set up the required roles for creating namespaces and managing resources in those namespaces; otherwise, you are restricted to a single address space. For more information about how to deploy without cluster-admin access, which will restrict {ProductName} to a single address space, see xref:deploying-single-address-space[].
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-address-space-controller.adoc]: NOTE doesn't need the * * around it.

Comment (ncbaratta) [reply documentation/service_admin/proc-deploying-address-space-controller.adoc]: cluster-admin is probably a literal

File: documentation/service_admin/proc-deploying-address-space-controller.adoc [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+----
+{cmdcli} create sa enmasse-admin
+----
+
+ifeval::["{cmdcli}" == "oc"]
+. Create cluster-wide roles used by the `enmasse-admin` service account:
++
+[options="nowrap"]
+----
+oc login -u system:admin
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-address-space-controller.adoc]: Instead of the 'if' why not just use the attribute in the code snippets?

File: documentation/service_admin/proc-deploying-address-space-controller.adoc [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+oc create -f ./resources/cluster-roles/openshift/address-space-controller.yaml
+----
+
+. Grant privileges to the service account:
++
+[options="nowrap"]
+----
+oc login -u system:admin
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-address-space-controller.adoc]: Why no attribute here of the 'oc' command but there is below?

File: documentation/service_admin/proc-deploying-api-server.adoc [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+include::common/attributes.adoc[]
+
+:cmdcli: oc
+
+[id='deploying-api-server_{context}']
+= (Optional) Deploying the API server
+
+The API server provides a REST API for creating address spaces and addresses. It can also serve as
+a Kubernetes API server if it is registered as an APIService.
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-api-server.adoc]: Kubernetes is only for the upstream, not the downstream.

File: documentation/service_admin/proc-deploying-api-server.adoc [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+*NOTE*: To install {ProductName} on OpenShift, you must have cluster-admin access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without cluster-admin access, which will restrict {ProductName} to a single address space, see xref:deploying-single-address-space[].
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-api-server.adoc]: cluster-admin is a literal

Comment (ncbaratta) [reply documentation/service_admin/proc-deploying-api-server.adoc]: Note doesn't needs the * *

File: documentation/service_admin/proc-deploying-api-server.adoc [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+----
+{cmdcli} create sa enmasse-admin
+----
+
+ifeval::["{cmdcli}" == "oc"]
+. Create cluster-wide roles used by the `enmasse-admin` service account:
++
+[options="nowrap"]
+----
+oc login -u system:admin
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-api-server.adoc]: Why not use the attribute instead of the if?

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+include::common/attributes.adoc[]
+
+:cmdcli: oc
+
+[id='deploying-auth-services_{context}']
+= Deploying authentication services
+
+{ProductName} requires at least one authentication service to be deployed. The authentication service
+can either be none (allow all), standard (keycloak), or external (not managed by {ProductName}).
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: none, standard, and external are all literals.

Comment (ncbaratta) [reply documentation/service_admin/proc-deploying-auth-services.adoc]: keycloak is a product name - capitalize the K

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+== Deploying the none authentication service
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: none is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+
+.Procedure
+
+. Create a certificate to use with the none authentication service. For testing purposes, you can create a self-signed certificate:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: none is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
++
+[options="nowrap"]
+----
+mkdir -p none-authservice-cert
+openssl req -new -x509 -batch -nodes -days 11000 -subj "/O=io.enmasse/CN=none-authservice.enmasse.svc.cluster.local" -out none-authservice-cert/tls.crt -keyout none-authservice-cert/tls.key
+----
+
+. Create a secret with the none authentication service certificate:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: none is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
++
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create secret tls none-authservice-cert --cert=none-authservice-cert/tls.crt --key=none-authservice-cert/tls.key
+----
+
+. Create the none authentication service:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: none is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
++
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create -f ./resources/none-authservice/service.yaml
+{cmdcli} create -f ./resources/none-authservice/deployment.yaml
+----
+
+== Deploying the standard authentication service
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+
+.Procedure
+
+. Create a certificate to use with the standard authentication service. For testing purposes, you can create a self-signed certificate:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
++
+[options="nowrap"]
+----
+mkdir -p standard-authservice-cert
+openssl req -new -x509 -batch -nodes -days 11000 -subj "/O=io.enmasse/CN=standard-authservice.enmasse.svc.cluster.local" -out standard-authservice-cert/tls.crt -keyout standard-authservice-cert/tls.key
+----
+
+. Create a secret with the standard authentication service certificate:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: standard is a literal

so is secret??

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
++
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create secret tls standard-authservice-cert --cert=standard-authservice-cert/tls.crt --key=standard-authservice-cert/tls.key
+----
+
+. Create a secret with keycloak admin credentials. Choose a password wisely as this user will have complete access over authentication and authorization policies:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: keycloak is a product name.  is secret a literal?

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+. Grant privileges to the service account:
++
+[options="nowrap"]
+----
+oc login -u system:admin
+oc adm policy add-cluster-role-to-user enmasse.io:keycloak-controller system:serviceaccount:enmasse:enmasse-admin
+----
+endif::[]
+
+. Create the standard authentication service:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-auth-services.adoc [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create -f ./resources/standard-authservice/service.yaml
+{cmdcli} create -f ./resources/standard-authservice/keycloak-deployment.yaml
+{cmdcli} create -f ./resources/standard-authservice/controller-deployment.yaml
+{cmdcli} create -f ./resources/standard-authservice/pvc.yaml
+{cmdcli} create -f ./resources/standard-authservice/route.yaml
+----
+
+. Create the keycloak configuration used by the controller and service. To make the standard authservice
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-auth-services.adoc]: keycloak is a product name or should this say standard?

Comment (ncbaratta) [reply documentation/service_admin/proc-deploying-auth-services.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-service-broker.adoc [hunk 26]
```diff
@@ -0,0 +1,119 @@
...
+include::common/attributes.adoc[]
+
+:cmdcli: oc
+
+[id='deploying-service-broker_{context}']
+= (Optional) Deploying the service broker
+
+The service broker provides an implementation of the Open Service Broker API that integrates with the Kubernetes Service Catalog. The service broker requires the standard authentication service to be deployed.
+
+*NOTE*: To install {ProductName} on OpenShift, you must have cluster-admin access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without cluster-admin access, which will restrict {ProductName} to a single address space, see xref:deploying-single-address-space[].
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: remove * *

File: documentation/service_admin/proc-deploying-service-broker.adoc [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+// assembly-installing-manual-steps.adoc
+
+include::common/attributes.adoc[]
+
+:cmdcli: oc
+
+[id='deploying-service-broker_{context}']
+= (Optional) Deploying the service broker
+
+The service broker provides an implementation of the Open Service Broker API that integrates with the Kubernetes Service Catalog. The service broker requires the standard authentication service to be deployed.
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: Kubernetes is upstream only

Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-service-broker.adoc (hunk 26 above)
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: cluster-admin is a literal

File: documentation/service_admin/proc-deploying-service-broker.adoc [hunk 28]
```diff
@@ -0,0 +1,119 @@
...
+
+*NOTE*: To install {ProductName} on OpenShift, you must have cluster-admin access to set up the required roles for delegating authentication to the Kubernetes master; otherwise, you are restricted to a single address space. For more information about how to deploy without cluster-admin access, which will restrict {ProductName} to a single address space, see xref:deploying-single-address-space[].
+
+.Prerequisite
+
+* The service broker requires the standard authentication service to be deployed.
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: standard is a literal

File: documentation/service_admin/proc-deploying-service-broker.adoc [hunk 29]
```diff
@@ -0,0 +1,119 @@
...
++
+[options="nowrap"]
+----
+oc login -u system:admin
+oc policy add-role-to-user admin system:serviceaccount:enmasse:enmasse-admin
+oc adm policy add-cluster-role-to-user enmasse.io:service-broker system:serviceaccount:enmasse:enmasse-admin
+oc adm policy add-cluster-role-to-user system:auth-delegator system:serviceaccount:enmasse:enmasse-admin
+----
++
+*Note*: You can log in again as the regular user after this step.
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: remove * *

File: documentation/service_admin/proc-deploying-service-broker.adoc [hunk 30]
```diff
@@ -0,0 +1,119 @@
...
+
+. Deploy the service broker:
++
+[options="nowrap",subs="attributes"]
+----
+{cmdcli} create -f ./resources/service-broker/deployment.yaml
+{cmdcli} create -f ./resources/service-broker/service.yaml
+----
+
+. To ensure the service broker behaves correctly in redirects, you must specify the `keycloakUrl` setting. If you are running a local cluster without a public DNS, use the internal service IP address for the host name; otherwise, use the host name of the external route. To obtain the service IP address, use this command:
```
Comment (ncbaratta) [inline documentation/service_admin/proc-deploying-service-broker.adoc]: "in redirects" is awkward.  "To ensure the service broker redirects correctly," might be better.

File: documentation/service_admin/proc-installing-using-ansible.adoc [hunk 31]
```diff
@@ -0,0 +1,32 @@
...
+//
+// assembly-installing-openshift.adoc
+
+include::common/attributes.adoc[]
+
+[id='installing-using-ansible_{context}']
+= Installing {ProductName} using Ansible
+
+Installing {ProductName} using Ansible requires creating an inventory file with the variables for configuring the
+system. Example inventory files can be found in the `ansible/inventory` folder. For more information about the supported Ansible configuration settings see xref:ref_ansible-config-settings_{enmasse}[Ansible configuration settings].
```
Comment (ncbaratta) [inline documentation/service_admin/proc-installing-using-ansible.adoc]: Remove the { } around enmasse 

xref:ref_ansible-config-settings_enmasse[Ansible configuration settings].

Comment (ncbaratta) [inline documentation/service_admin/proc-installing-using-ansible.adoc]: directory instead of folder

File: documentation/service_admin/installing-openshift.adoc (hunk 3 above)
Comment (jenmalloy) [reply documentation/service_admin/installing-openshift.adoc]: Fixed

File: documentation/common/snip-prerequisites-openshift.adoc (hunk 6 above)
Comment (jenmalloy) [reply documentation/common/snip-prerequisites-openshift.adoc]: I'm not so sure about that, since {ProductName} is there. But it's probably not Origin that we want for downstream? @rgodfrey, do you know?

File: documentation/service_admin/proc-deploying-address-space-controller.adoc (hunk 9 above)
Comment (jenmalloy) [reply documentation/service_admin/proc-deploying-address-space-controller.adoc]: Done

File: documentation/service_admin/proc-installing-using-ansible.adoc (hunk 31 above)
Comment (jenmalloy) [reply documentation/service_admin/proc-installing-using-ansible.adoc]: Fixed

File: documentation/service_admin/assembly-installing-openshift.adoc (hunk 7 above)
Comment (jenmalloy) [reply documentation/service_admin/assembly-installing-openshift.adoc]: Fixed

File: documentation/service_admin/proc-deploying-auth-services.adoc (hunk 15 above)
Comment (jenmalloy) [reply documentation/service_admin/proc-deploying-auth-services.adoc]: Fixed

File: documentation/service_admin/proc-deploying-auth-services.adoc (hunk 22 above)
Comment (jenmalloy) [reply documentation/service_admin/proc-deploying-auth-services.adoc]: Don't think secret is a literal. It's like a password

File: documentation/common/snip-prerequisites-openshift.adoc [hunk 32]
```diff
@@ -1,4 +1,4 @@
-:context: enmasse
+:context: maas
```
Comment (ncbaratta) [inline documentation/common/snip-prerequisites-openshift.adoc]: We're only going to want context defined in master.adoc unless it's when including a file we plan to reuse. Then we want it above the include statement. As this is a snippet, defining the context here will cause some big issues.

Comment (jenmalloy) [reply documentation/common/snip-prerequisites-openshift.adoc]: Fixed for both snip files

File: documentation/common/snip-create-openshift-project.adoc [hunk 33]
```diff
@@ -1,4 +1,4 @@
-. Create the `enmasse` project:
+* Create the `enmasse` project:
```
Comment (ncbaratta) [inline documentation/common/snip-create-openshift-project.adoc]: Does it have to be enmasse?
